### ISSUE TITLE
IMG tag embedded in token notes doesn't render without height and width attrs.

### ISSUE THREAD

**Issue by jschoudt** [RPTools/maptool#2549]
**Describe the bug**
When you insert a `<img src='asset://...'/>` into the notes or GM notes of an object in the object layer, it does not render at the image's native size. I suspect it's using 0 for the width and height of the image, or the html engine isn't doing the typical second pass that re-renders images at their native size once the image data is loaded asynchronously.

I've used this in this campaign in maptool 1.6.x and 1.7.x with no width/height attrs and it worked fine. This was my way of providing a handout to my players that they could see at will. I would put these markers on the game overview map and the players could then refer back to it to review handouts or pictures from a particular location in my game very conveniently.

**To Reproduce**
1. Drag an image to the object layer of a map
2. Create a macro 'foo' with the code:
```
[h: "<!-- Pulling the image asset ID and building the string that will work in the notes section of the token. -->"]
[h: assetURL = getTokenImage()]
[h: imageText = "<br><img src='"+assetURL+"'/>"]
[h: setNotes(imageText)]
```
3. Run the macro
4. Go to the notes section of the object config and ensure that an img tag is there.
4. Switch to token layer, click the object. You will see a blank notes panel popup.
5. Go back to the object layer, edit the object notes and insert a width and height attribute on the image tag like: `<br><img height=100 width=100 src='asset://7b695a053b8f55de4ddd02307b934af4'/>`
6. Switch back to token layer, click the object and you'll see the panel popup with a 100x100 rendering of the image you dragged.

**Expected behavior**
Inserting an img tag with the src referring to an asset:// url in the notes or GM notes of an object in the object layer should then display the referenced image in the popup panel when you click the object while in the token layer. This worked prior to 1.8.x.

**Screenshots**
Create an object in the object layer with an img tag referring to an asset:// url:
![object-with-img-tag-no-size](https://user-images.githubusercontent.com/1143633/115127480-c7c59600-9fa4-11eb-874e-3a24c6147044.jpg)

Switch to token layer and click the object, popup is empty.
![object-with-img-tag-no-size-popup](https://user-images.githubusercontent.com/1143633/115127490-d9a73900-9fa4-11eb-8a09-ea7b84ba4980.jpg)

Go back to object layer, edit the object and insert width and height attrs:
![object-with-img-tag-with-size](https://user-images.githubusercontent.com/1143633/115127500-e62b9180-9fa4-11eb-9c97-c892cc798aaf.jpg)

Switch to token layer and click the object, img is in the popup:
![object-with-img-tag-with-size-popup](https://user-images.githubusercontent.com/1143633/115127507-f3e11700-9fa4-11eb-8fb3-d2dc1bb98ec6.jpg)

**MapTool Info**
- Version: 1.8.3, 1.8.4, 1.8.5
- Install: Removed 1.7.x and installed 1.8.3. Then upgraded to 1.8.5 before I noticed that this was happening. Then I uninstalled 1.8.5 and re-installed 1.8.4 to test there, and then uninstalled/re-installed 1.8.3 to test there.

**Desktop (please complete the following information):**
 - OS: Windows
 - Version: 10 Home 19042.928

**Additional context**
Gather Debug information:
```
==== MapTool Information ====
MapTool Version: 1.8.3
MapTool Home...: C:\Users\jschoudt\.maptool-rptools
MapTool Install: C:\Users\jschoudt\AppData\Local\MapTool1.8.3\app
Max mem avail..: 7 GB
Max mem used...: 2 GB
Custom Property: -DMAPTOOL_DATADIR=.maptool-rptools

==== Java Information ====
Java Home......: C:\Users\jschoudt\AppData\Local\MapTool1.8.3\runtime
Java Vendor....: AdoptOpenJDK
Java Version...: 14.0.2
Java Parameters: 
  -Djava.library.path=C:\Users\jschoudt\AppData\Local\MapTool1.8.3\app;C:\Users\jschoudt\AppData\Local\MapTool1.8.3
  -Djava.launcher.path=C:\Users\jschoudt\AppData\Local\MapTool1.8.3
  -Xss8M
  -Dsentry.environment=Production
  -Dfile.encoding=UTF-8
  -XX:+ShowCodeDetailsInExceptionMessages
  --add-opens=javafx.web/javafx.scene.web=ALL-UNNAMED
  --add-opens=javafx.web/com.sun.webkit=ALL-UNNAMED


==== OS Information ====
OS Name........: Windows 10
OS Version.....: 10.0
OS Architecture: amd64
Processor......: AMD64 Family 23 Model 113 Stepping 0, AuthenticAMD
PATH...........: C:\Users\jschoudt\AppData\Local\MapTool1.8.3\app;C:\Users\jschoudt\AppData\Local\MapTool1.8.3
Number of Procs: 16

==== User Information ====
User Name: jschoudt
User Home: C:\Users\jschoudt
User Dir.: C:\Users\jschoudt\AppData\Local\MapTool1.8.3

==== Network Interfaces ====
Display Name..: Software Loopback Interface 1
Interface Name: lo
Address...: 127.0.0.1
Address...: 0:0:0:0:0:0:0:1

Display Name..: Intel(R) I211 Gigabit Network Connection
Interface Name: eth1
Address...: 192.168.1.23
Address...: fe80:0:0:0:e8a6:49db:534e:62da%eth1

Display Name..: Hyper-V Virtual Ethernet Adapter
Interface Name: eth16
Address...: 172.28.64.1
Address...: fe80:0:0:0:f464:154c:ac5a:e455%eth16

Host Address...: 192.168.1.23
Default Gateway: 192.168.1.1

==== Locale Information ====
Country.: United States
Language: English
Locale..: English (United States)
Variant.: 

==== Encoding Information ====
Default Locale:   en_US
Default Charset:  UTF-8
file.encoding:    UTF-8
sun.jnu.encoding: Cp1252
Default Encoding: UTF8

==== Display Information ====
Number of Displays: 2
Display 1: 3840x2160(32)
Display 2: 1920x1080(32)

==== Internet Gateway Devices ====
Device Name.: RBR50 (Gateway)
Model Name..: NETGEAR Orbi Desktop AC3000 Router
Manufacturer: NETGEAR, Inc.
Model Number: RBR50
Model Desc..: http://www.netgear.com/home/products/wirelessrouters
Firmware....: Netgear_Orbi UPnP/1.0 miniupnpd/1.0
External IP.: 24.247.182.26
```

**Comment by Phergus** [comment]
Odd.  I'm surprised no one else noticed this.  Possibly an issue with the change to Java 14 for 1.8.x.  With current develop code and 1.9.0-alpha.9 the image tag works fine without using width/height attributes.

Is there anything in your MapTool logs?

**Comment by Phergus** [comment]
I was the same way.  I had to check that it did work in older versions.  I typically just throw an image in the portrait for the object.

**Comment by jschoudt** [comment]
Yeah, the portrait works for smaller images. But for larger things with associated text flowing with it this was a better solution for me than handouts.

**Comment by jschoudt** [comment]
If it's fixed in 1.9.x, feel free to close this. I implemented a workaround where I pull the asset properties and specify w and h in the img tag now.

**Comment by Phergus** [comment]
Being able to embed them in the notes properly is certainly a good thing. Fortunately it's fixed in 1.9.

Closing.
